### ISSUE TITLE
AttributeError: /usr/lib/libcapstone.so: undefined symbol: cs_disasm

### ISSUE THREAD

**Issue by bprykhodchenko** [Veil-Framework/Veil-Evasion#370]
## Veil-Evasion version
Veil-Evasion | [Version]: 2.27.2

## OS Used - all info (architecture, linux flavor, etc)

3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt25-2+deb8u2 (2016-06-25) x86_64 GNU/Linux

## Pastebin link to error you are encountering (include console actions you took prior to error)

Traceback (most recent call last):
  File "./Veil-Evasion.py", line 325, in <module>
    controller = controller.Controller(oneRun=False)
  File "/opt/Veil-Evasion/modules/common/controller.py", line 107, in __init__
    self.LoadPayloads()
  File "/opt/Veil-Evasion/modules/common/controller.py", line 119, in LoadPayloads
    d = dict( (path[path.find("payloads")+9:-3], imp.load_source( "/".join(path.split("/")[3:])[:-3],path )  ) for path in glob.glob(join(settings.VEIL_EVASION_PATH+"/modules/payloads/" + "*/" * x,'[!_]*.py')) )
  File "/opt/Veil-Evasion/modules/common/controller.py", line 119, in <genexpr>
    d = dict( (path[path.find("payloads")+9:-3], imp.load_source( "/".join(path.split("/")[3:])[:-3],path )  ) for path in glob.glob(join(settings.VEIL_EVASION_PATH+"/modules/payloads/" + "*/" * x,'[!_]*.py')) )
  File "/opt/Veil-Evasion//modules/payloads/native/backdoor_factory.py", line 15, in <module>
    from tools.backdoor import pebin
  File "/opt/Veil-Evasion/tools/backdoor/pebin.py", line 45, in <module>
    from intel.intelCore import intelCore
  File "/opt/Veil-Evasion/tools/backdoor/intel/intelCore.py", line 38, in <module>
    from capstone import *
  File "/usr/local/lib/python2.7/dist-packages/capstone/__init__.py", line 286, in <module>
    ctypes.c_uint64, ctypes.c_size_t, ctypes.POINTER(ctypes.POINTER(_cs_insn)))
  File "/usr/local/lib/python2.7/dist-packages/capstone/__init__.py", line 281, in _setup_prototype
    getattr(lib, fname).restype = restype
  File "/usr/lib/python2.7/ctypes/__init__.py", line 378, in __getattr__
    func = self.__getitem__(name)
  File "/usr/lib/python2.7/ctypes/__init__.py", line 383, in __getitem__
    func = self._FuncPtr((name_or_ordinal, self))
AttributeError: /usr/lib/libcapstone.so: undefined symbol: cs_disasm
## Any additional info you want to tell me

After installation and executing a script sudo ./Veil-evasion it gives me the output above. I cannot even launch it...

**Comment by ChrisTruncer** [comment]
Did you run the setup with "sudo ./setup.sh -c"?  Could you try reinstalling? I'm running on Debian right now and have never seen this issue before

**Comment by bprykhodchenko** [comment]
Yeap, I have tried to reinstall it a couple of times, but nothing helped.

Also, maybe it will help. In the installation process I see that
Downloading http://pypi.python.org/packages/source/d/distribute/distribute-0.6.14.tar.gz
Traceback (most recent call last):
  File "distribute_setup.py", line 485, in <module>
    main(sys.argv[1:])
  File "distribute_setup.py", line 480, in main
    tarball = download_setuptools()
  File "distribute_setup.py", line 197, in download_setuptools
    dst = open(saveto, "wb")
IOError: [Errno 13] Permission denied: 'Z:\\opt\\Veil-Evasion\\setup\\distribute-0.6.14.tar.gz'

But even if I install it via root user it gives me the same result...
So I don't think the problem is in permissions, it seems like capstone don't work correctly...

**Comment by veil-evasion** [comment]
That is definitely a permissions issue

**Comment by bprykhodchenko** [comment]
Well, I have figured out what is the problem.
For that I have even completely reinstalled my Debian on my Laptop.
So, I have noticed, that installing on a fresh system goes without any issues.
Well, ok, Veil has started working, but next I have installed MItM framework and I had the same issue as described above.
Even my MITM framework gave me the same error.
I have decided to explore deeper and started googling.
Finally! The problem was that capstone that comes with default aptitude and pip is the 2th version. But for the proper working needed capstone is 3th version.
So I have removed my capstone2 completely and downloaded and compiled capstone 3 from the official web site. And that is it! All wors like a charm.
Hope this will help to somebody having the same trouble.
